src/gluonnlp/data/tokenizers.py

```diff
@@ -0,0 +1,1715 @@
+            default_special_tokens = {'pad_token':self._pad_token,
+                                      'cls_token':self._cls_token,
+                                      'sep_token':self._sep_token,
+                                      'mask_token':self._mask_token}
+            self._vocab = Vocab(all_tokens, unk_token=self._unk_token, **default_special_tokens)
+            all_tokens = self._vocab.all_tokens
+        # for safety, also use temp file when using wordpiece vocab file
+        # for situation that original all_tokens not cotain special tokens
+        # (vocab file of BERT do not contain all special tokens)
+        temp_hf_vocab_file = str(uuid4()) + '.hf_vocab'
+        with open(temp_hf_vocab_file, 'w', encoding='utf-8') as ftv:
+            ftv.write('\n'.join(all_tokens))
+        self._vocab.mask_token_id = self._vocab.mask_id
+        assert [self._unk_token, self._sep_token, self._cls_token, self._pad_token, self._mask_token] == \
+               [self._vocab.unk_token, self._vocab.sep_token, self._vocab.cls_token,
+                self._vocab.pad_token, self._vocab.mask_token]
+        self._bpe = tokenizers.BertWordPieceTokenizer(
+            vocab_file=temp_hf_vocab_file if temp_hf_vocab_file else self._vocab_file,
+            unk_token=self._unk_token,
+            sep_token=self._sep_token,
+            cls_token=self._cls_token,
+            pad_token=self._pad_token,
+            mask_token=self._mask_token,
+            clean_text=self._clean_text,
+            handle_chinese_chars=self._handle_chinese_chars,
+            strip_accents=self._strip_accents, lowercase=self._lowercase,
+            wordpieces_prefix=self._wordpieces_prefix)
+        os.remove(temp_hf_vocab_file)
+
+    def __repr__(self):
+        ret = '{}(\n' \
+              '   vocab_file = {}\n' \
+              '   unk_token = {}, sep_token = {}, cls_token = {}\n' \
+              '   pad_token = {}, mask_token = {}\n' \
+              '   clean_text = {}, handle_chinese_chars = {}\n' \
+              '   strip_accents = {}, lowercase = {}\n' \
+              '   wordpieces_prefix = {}\n' \
+              '   vocab = {}\n' \
+              ')'.format(self.__class__.__name__,
+                         os.path.realpath(self._vocab_file),
+                         self._unk_token, self._sep_token, self._cls_token,
+                         self._pad_token, self._mask_token,
+                         self._clean_text, self._handle_chinese_chars,
+                         self._strip_accents, self._lowercase,
+                         self._wordpieces_prefix,
+                         self._vocab)
+        return ret
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state['_bpe'] = None
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        self.__rebuild_tokenizer()
+
+
+@TOKENIZER_REGISTRY.register('spm')
+class SentencepieceTokenizer(BaseTokenizerWithVocab):
```

Comment:
Can we break it down into multiple files, each for 1 tokenizer?

Reply:
Let me do it in a separate PR.